gulpfile.js

```diff
@@ -72,21 +80,83 @@ gulp.task(
+  return depcheck(__dirname, {
+    // "@types/*" dependencies are type declarations that are automatically
+    // loaded by TypeScript during build. depcheck can't detect this
+    // so we ignore them here. Same with babel plugins.
+    ignoreMatches: [
+      '@types/*', 'vinyl', 'babel-plugin-external-helpers']
+  }).then((result) => {
+    let invalidFiles = Object.keys(result.invalidFiles) || [];
+    let invalidJsFiles = invalidFiles.filter((f) => f.endsWith('.js'));
+    if (invalidJsFiles.length > 0) {
+      throw new Error(`Invalid files: ${invalidJsFiles}`);
+    }
+    if (result.dependencies.length) {
+      throw new Error(`Unused dependencies: ${result.dependencies}`);
+    }
+  });
+});
+
+/*
+ * There doesn't seem to be documentation on what helpers are available, or
+ * which helpers are required for which transforms. The
+ * source is here: https://github.com/babel/babel/blob/6.x/packages/babel-helpers/src/helpers.js
+ *
+ * This list is an educated guess at the helpers needed for our transform set
+ * of ES2015 - modules. When we switch to loose mode we should update the list.
+ *
+ * All helpers are listed here, with some commented out, so it's clear what we've excluded.
+ */
+const babelHelperWhitelist = [
+  'typeof', // Symbol support, for IE11
+  // 'jsx', // we don't support JSX
+  // 'asyncIterator', // async-iterators are not in ES2015
+  // 'asyncGenerator', // async-iterators are not in ES2015
+  // 'asyncGeneratorDelegate', // async-iterators are not in ES2015
+  // 'asyncToGenerator', // async functions are not in ES2015
+  'classCallCheck',
+  'createClass',
+  'defineEnumerableProperties',
+  'defaults', // TODO (justinfagnani): what's this for?
+  'defineProperty',
+  'extends',
+  'get', // not needed? Seems to implement getters
+  'inherits', // TODO (justinfagnani): what's the difference from 'extends'?
+  'instanceof',
+  // 'interopRequireDefault', // for modules support
+  // 'interopRequireWildcard', // for modules support
+  'newArrowCheck', // Can we exclude with loose?
+  'objectDestructuringEmpty',
+  'objectWithoutProperties',
+  'possibleConstructorReturn', // can we exclude with loose?
+  // 'selfGlobal', // not needed? global is not ES2015
+  'set', // not needed? Seems to implement setters
+  'slicedToArray',
+  // 'slicedToArrayLoose',
+  'taggedTemplateLiteral',
+  // 'taggedTemplateLiteralLoose',
+  'temporalRef', // not needed in loose?
+  'temporalUndefined',
+  'toArray',
+  'toConsumableArray',
+];
+
+gulp.task('gen-babel-helpers', () => {
+  return Promise.resolve().then(() => {
+    return streamToPromise(pump(
+      run(
+      `babel-external-helpers --whitelist ${babelHelperWhitelist.join(',')}`,
+      { silent: true })
+      .exec(),
+      rename('babel-helpers.js'),
+      gulp.dest('./lib/build/')
+    ));
+  }).then(() => {
+    return streamToPromise(pump(
+      run('uglifyjs ./lib/build/babel-helpers.js -m', { silent: true }).exec(), rename('babel-helpers.min.js'),
```

Comment:
Same here, uglify has a javascript API as well. Use that instead. We could get rid of `gulp-run` and `gulp-rename` completely that way.